### ISSUE TITLE
attempt to index nil value error when trying to create a directory/file 

### ISSUE THREAD

**Issue by gozes** [nvim-telescope/telescope-file-browser.nvim#7]
When I try to create a directory and a file In folder mode I get the error on the screens shot below. worth pointing out that if I'm in file mode and hit `/` it will put me in folder mode so it seems to me that the only way to that this plugging has for creating a file may be a bit broken.

 
<img width="852" alt="Screenshot 2021-11-21 at 12 29 15" src="https://user-images.githubusercontent.com/632922/142762069-00c6731e-e90d-4158-b24b-015e9b1a4ed1.png">

**Comment by fdschmidt93** [comment]
Couple of things:

* Proper steps to repro would be much appreciated -- thanks for reminding me that I have to put up an issue template
* Creating a file just works differently than the file browser from telescope core because this extension uses `vim.ui.input`. You just press `<C-e>` and then `vim.ui.input` is triggered and works for me as expected in file and folder browser mode
* Apologies, `/` is a currently undocumented input callback to move to folder browser mode as suggested upstream on the original PR. 

> worth pointing out that if I'm in file mode and hit / it will put me in folder mode so it seems to me that the only way to that this plugging has for creating a file may be a bit broken.

As mentioned in first thing on the README, this project is currently unstable. The file browser from core will only be deprecated in 19th of December. Hence, I'd welcome more constructive issues, rather than a vague description of _what exactly_ it is you are doing in an expecting manner.

I'm closing this because I've retested it a couple of times and from what's described, it works as intended. Please let me know in case the above description doesn't work.

**Comment by j-hui** [comment]
@fdschmidt93 I just ran into that issue as well. It looks like `vim.ui` is not available on Neovim v0.5.1. I tested by running the following:

```viml
lua vim.ui.input({prompt = 'type: '}, function(i) print('typed: '..i) end)
```

This throws the exact same error on 0.5.1, but works as expected on another machine where I happen to have Neovim 0.6.0.

I'm personally looking forward to 0.6.0 to be released soon (or will just fix my machine running 0.5.1), but if the dependency on 0.6.0 is intentional, perhaps a warning could be given in the README?

**Comment by fdschmidt93** [comment]
Many thanks for the constructive feedback!

Apologies, I'm on nightly and didn't keep in mind this is not part of 0.5.1 yet. Nevertheless, this should be quite easily fixed up until `nvim 0.6` is released, which shouldn't be too far out.

Could you please confirm that

```lua
x = vim.ui and vim.ui input or vim.fn.input
print(x)
```

prints "<function 1>" (not sure about number) for you?

Then I'll push an easy intermediate fix.

**Comment by fdschmidt93** [comment]
https://neovim.discourse.group/t/neovim-0-6-stabilization-period-begins-today-10-31-2021/1329

Ok, I'm more leaning to not fix this, because the file browser will be removed mid December from telescope by when `0.6` will be hopefully released.

Given the quite significant difference in implementation, I'd have to rewrite some of the actions, which is too much time for me right now for something that'll fix itself too soon enough I'm afraid. I realize it is not satisfying for 0.5.1 users but I have to be practical with the time I have right now. My sincerest apologies.

I'll make a disclaimer on the README and would be happy to accept a PR to have a compat branch or so.

**Comment by j-hui** [comment]
For what it's worth @fdschmidt93 this works

```lua
x = vim.ui and vim.ui.input or vim.fn.input
print(x)
```

No worries about not having the fix for 0.5! I'm still a new user of this plugin (and the entire Telescope ecosystem), and as you said 0.6 should be coming around pretty soon (: appreciate you looking into this

**Comment by benfrain** [comment]
I get the same error on NVIM v0.6.0-dev+2040-g8cbe100fc if I type any nonsense in the file browser text area (so a string that doesn't match a file/folder) and press enter:

More specifically:

```
E5108: Error executing lua ...r.nvim/lua/telescope/_extensions/file_browser/picker.lua:82: attempt to inde
x a nil value
stack traceback:
        ...r.nvim/lua/telescope/_extensions/file_browser/picker.lua:82: in function 'condition'
        ...k/packer/opt/telescope.nvim/lua/telescope/actions/mt.lua:8: in function 'run_replace_or_origina
l'
        ...k/packer/opt/telescope.nvim/lua/telescope/actions/mt.lua:30: in function 'run_replace_or_origin
al'
        ...k/packer/opt/telescope.nvim/lua/telescope/actions/mt.lua:30: in function 'key_func'
        ...ack/packer/opt/telescope.nvim/lua/telescope/mappings.lua:235: in function 'execute_keymap'
        [string ":lua"]:1: in main chunk
```
